frontend/src/hooks/useFetchUser.ts

```diff
@@ -0,0 +1,21 @@
+import { useQuery } from "@tanstack/react-query";
+import useDecodeToken from "hooks/useDecodeToken";
+import { getUser } from "services/UsersService";
+
+const useFetchUser = () => {
+    const { id, userToken } = useDecodeToken();
+
+    const {
+        data: user,
+        isInitialLoading,
+        error
+    } = useQuery({
+        queryKey: ["user", id, userToken],
+        queryFn: () => getUser(id),
+        enabled: !!id && id !== 0 && !!userToken
```

Comment:
`!!id` will be `true` when `id` is greater than 0 so first boolean is enough I belive.

Reply:
"!!" operator check only if the value is not "null"

https://stackoverflow.com/questions/7452720/what-does-the-double-exclamation-operator-mean

Reply:
Yes, so it's still working with `0`. But it's more secure like you did because negative numbers are truthy, so leave it as it is.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean